# HG changeset patch
# User robertm@google.com
# Date 1298645669 18000
# Branch pnacl-sfi
# Node ID 6ed4ada37f726191f5dbc8e1ab6dfc2765e301d9
# Parent 543de3b563e304be2f6f457475d0d44e1d25f46c
sfi_data_tst was not available when FlagSfiZeroMask == true

This is needed for http://codereview.chromium.org/6574016
which added a crude way to test pnacl shared libs even if we do not have a properly sandboxed ld.so

review: http://codereview.chromium.org/6576046/

 From llvm-pnacl-0012-232-6ed4ada37f726191f5dbc8e1ab6dfc2765e301d9.patch

 From .hg/patches/llvm-pnacl-0012-232-6ed4ada37f726191f5dbc8e1ab6dfc2765e301d9.patch.stripped

diff -r 543de3b563e3 lib/Target/ARM/ARMNaClHeaders.cpp
--- a/lib/Target/ARM/ARMNaClHeaders.cpp	Wed Feb 23 15:36:34 2011 -0800
+++ b/lib/Target/ARM/ARMNaClHeaders.cpp	Wed Jun 29 23:04:56 2011 -0700
@@ -69,6 +69,9 @@
 
   O << " @ ========================================\n";
   if (FlagSfiZeroMask) {
+    // This mode sets all mask to zero which makes them into nops
+    // this is useful for linking this code against non-sandboxed code
+    // for debugging purposes
     O <<
       "\t.macro sfi_data_mask reg cond\n"
       "\tbic\\cond \\reg, \\reg, #0\n"
@@ -76,6 +79,12 @@
       "\n\n";
 
     O <<
+      "\t.macro sfi_data_tst reg\n"
+      "\ttst \\reg, #0x00000000\n"
+      "\t.endm\n"
+      "\n\n";
+
+    O <<
       "\t.macro sfi_code_mask reg cond=\n"
       "\tbic\\cond \\reg, \\reg, #0\n"
       "\t.endm\n"
